File: 프로그래머스/lv1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
class Solution {
    public int solution(String s) {
        int idx = 0;
        String str = "";
        while(idx < s.length()) {
            String tmp = "";
            while(!(s.charAt(idx) >= '0' && s.charAt(idx) <= '9')) {
                tmp += s.charAt(idx);
                idx++;
                if(tmp.equals("zero")) {
                    str += "0";
                    break;
                } else if(tmp.equals("one")) {
                    str += "1";
                    break;
                }  else if(tmp.equals("two")) {
                    str += "2";
                    break;
                }  else if(tmp.equals("three")) {
                    str += "3";
                    break;
                }  else if(tmp.equals("four")) {
                    str += "4";
                    break;
                }  else if(tmp.equals("five")) {
                    str += "5";
                    break;
                }  else if(tmp.equals("six")) {
                    str += "6";
                    break;
                }  else if(tmp.equals("seven")) {
                    str += "7";
                    break;
                }  else if(tmp.equals("eight")) {
                    str += "8";
                    break;
                }  else if(tmp.equals("nine")) {
                    str += "9";
                    break;
                }
            }
            if(tmp.equals("")) {
                str += s.charAt(idx);
                idx++;
            }
        }
        return Integer.parseInt(str);
    }
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    var answer = '';
    const map = new Map([
      ["zero", '0'],
      ["one", '1'],
      ["two", '2'],
      ["three", '3'],
      ["four", '4'],
      ["five", '5'],
      ["six", '6'],
      ["seven", '7'],
      ["eight", '8'],
      ["nine", '9'],
    ]);
    let start = 0;
    while(start < s.length) {
        if(s[start] >= '0' && s[start] <= '9') answer += s[start++]
        else {
            var str = '';
            while(start < s.length) {
                str += s[start];
                start++;
                if(map.has(str)) {
                    answer += map.get(str);
                    break;
                }
            }
        }
    }
    return Number(answer);
}